hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/TestCloseContainer.java

```diff
@@ -116,8 +129,18 @@ public void testReplicasAreReportedForClosedContainerAfterRestart()
     // Pick any container on the cluster, get its pipeline, close it and then
     // wait for the container to close
     ContainerInfo container = scm.getContainerManager().getContainers().get(0);
+    // Checksum file doesn't exist before container close
+    List<HddsDatanodeService> hddsDatanodes = cluster.getHddsDatanodes();
+    for (HddsDatanodeService hddsDatanode: hddsDatanodes) {
+      assertFalse(containerChecksumFileExists(hddsDatanode, container));
+    }
     OzoneTestUtils.closeContainer(scm, container);
 
+    // Checksum file exists after container close
+    for (HddsDatanodeService hddsDatanode: hddsDatanodes) {
+      GenericTestUtils.waitFor(() -> waitForContainerChecksumToExist(hddsDatanode, container), 100, 5000);
+    }
```

Comment:
`OzoneTestUtils#closeContainer` waits for the container to close on SCM. It would be better to explicitly wait for the container to close on the datanode here, then check that the checksum file exists. This ensures that the file is written synchronously on close, which is a stronger guarantee than the test currently provides.